server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java

```diff
@@ -578,15 +577,6 @@ private void fetchScans() throws Exception {
         ThriftUtil.returnClient(tserver);
       }
     }
-    // Age off old scan information
-    Iterator<Entry<HostAndPort,ScanStats>> entryIter = allScans.entrySet().iterator();
-    long now = System.currentTimeMillis();
-    while (entryIter.hasNext()) {
-      Entry<HostAndPort,ScanStats> entry = entryIter.next();
-      if (now - entry.getValue().fetched > 5 * 60 * 1000) {
-        entryIter.remove();
-      }
-    }
```

Comment:
I don't think this change will result in desired behavior. There is no other code that will ever remove anything from `allScans` other than this code. So, that object will continue to grow indefinitely, keeping every scan ever fetched, until the monitor falls over due to memory problems.

Reply:
The map will only grow per tserver. For each tserver, we get the active scans and do a straight put in the map:
<pre>
allScans.put(parsedServer, new ScanStats(scans));
</pre>

So yes it could grow as tservers are added and removed from the cluster but not by that much. Either way, this could be done better so it only stores the relevant data.

I am trying to simulate a slow running scan in Uno using `SlowIterator`. If I can get that to work, I will make this better.